core/src/main/scala/com/rallyhealth/vapors/factfilter/evaluator/InterpretExprAsFunction.scala

```diff
@@ -0,0 +1,504 @@
+    expr.visit(InterpretExprAsFunction())(input)
+  }
+
+  final case class Input[F[_], V](
+    value: F[V],
+    evidence: ResultSet,
+    factTable: FactTable,
+  ) {
+
+    @inline def withFoldableValue[G[_], U](value: G[U]): Input[G, U] = copy(value = value)
+
+    @inline def withValue[U](value: U): Input[Id, U] = withFoldableValue[Id, U](value = value)
+  }
+
+  final object Input {
+
+    type Init = Input[Id, FactTable]
+
+    @inline def fromFactTable(factTable: FactTable): Init =
+      Input[Id, FactTable](factTable, NoFactsMatch(), factTable)
+
+    @inline def fromValue[V](
+      value: V,
+      evidence: ResultSet,
+      factTable: FactTable,
+    ): Input[Id, V] =
+      Input[Id, V](value, evidence, factTable)
+
+    @inline def fromValue[V](
+      value: V,
+      evidence: ResultSet = NoFactsMatch,
+    ): Input[Id, V] =
+      fromValue(value, evidence, FactTable(evidence.matchingFacts))
+  }
+
+  final case class Output[R](
+    value: R,
+    evidence: ResultSet,
+  )
+
+  object Output {
+
+    /**
+      * Logical Conjunction (aka AND)
+      *
+      * - If either side is false, the result is false
+      * - Evidence of falseness is accumulated
+      * - Evidence of truthiness requires evidence of truth on both sides, otherwise no evidence of truth
+      *
+      * Examples:
+      *
+      * | X is True | Evidence of X | Y is True | Evidence of Y | Result is True | Evidence of Result |
+      * | --------- | ------------- | --------- | ------------- | -------------- | ------------------ |
+      * | T         | {}            | T         | {}            | T              | {}                 |
+      * | T         | {}            | F         | {}            | F              | {}                 |
+      * | T         | {}            | T         | {A}           | T              | {}                 |
+      * | T         | {}            | F         | {A}           | F              | {A}                |
+      * | T         | {B}           | T         | {A}           | T              | {A, B}             |
+      * | T         | {B}           | F         | {}            | F              | {}                 |
+      * | T         | {B}           | T         | {}            | T              | {}                 |
+      * | T         | {B}           | F         | {A}           | F              | {A}                |
+      * | F         | {}            | T         | {A}           | F              | {}                 |
+      * | F         | {B}           | F         | {A}           | F              | {A, B}             |
+      * | F         | {B}           | T         | {A}           | F              | {B}                |
+      * | F         | {B}           | F         | {}            | F              | {B}                |
+      * | F         | {}            | F         | {A}           | F              | {}                 |
```

Comment:
I feel like I need to study this file some more. I'm also still a little questionable about results without evidence.

Reply:
With regards to results without evidence, as long as no operator creates new evidence in an unwarrented fashion, I think it is up to the receiver of the result to interpret a result without evidence.